lib/Dialect/FIRRTL/FIRRTLFolds.cpp

```diff
@@ -38,6 +38,15 @@ static Value dropWrite(PatternRewriter &rewriter, OpResult old,
   return passthrough;
 }
 
+static Value moveNameHint(OpResult old, Value passthrough) {
+  Operation *op = passthrough.getDefiningOp();
```

Comment:
I think this is going to not work correctly for ports?

Reply:
It doesn't need to.